### PR TITLE
ssh: compare hostname with a regex 

### PR DESCRIPTION
**Pull request by eduardoveiga**

### CODE REVIEW

File: api/routes/sshkeys.go [hunk 1]
```diff
@@ -128,3 +129,24 @@ func CreatePrivateKey(c apicontext.Context) error {
...
+	pubKey, err := svc.GetPublicKey(c.Ctx(), c.Param("fingerprint"), c.Param("tenant"))
+	if err != nil {
+		return c.JSON(http.StatusForbidden, err)
+	}
+
+	if err := c.Bind(&device); err != nil {
+		return c.JSON(http.StatusForbidden, err)
+	}
+
+	ok, err := svc.EvaluateKeyHostname(c.Ctx(), pubKey, device)
```
Comment (otavio) [inline api/routes/sshkeys.go:146]: You can move the `device` declaration just before its use here.

Comment (otavio) [inline api/routes/sshkeys.go:146]: Check `err` and take `ok` as default `return` if all previous checks work.

File: pkg/api/client/client_internal.go [hunk 2]
```diff
@@ -42,6 +43,21 @@ func (c *client) GetPublicKey(fingerprint, tenant string) (*models.PublicKey, er
 	return pubKey, nil
 }
 
+func (c *client) EvaluateKey(fingerprint string, dev *models.Device) (bool, error) {
+
+	resp, _, errs := c.http.Post(buildURL(c, fmt.Sprintf("/internal/sshkeys/public-keys/evaluate/%s", fingerprint))).Send(dev).End()
+
```
Comment (otavio) [inline pkg/api/client/client_internal.go:49]: drop the empty lines.

File: ssh/server.go [hunk 3]
```diff
@@ -244,10 +242,15 @@ func (*Server) publicKeyHandler(ctx sshserver.Context, pubKey sshserver.PublicKe
 
 	if ssh.FingerprintLegacyMD5(magicPubKey) != fingerprint {
 		apiClient := client.NewClient()
-		_, err = apiClient.GetPublicKey(fingerprint, device.Tenant)
+		_, err = apiClient.GetPublicKey(fingerprint, device.TenantID)
 		if err != nil {
 			return false
 		}
+
+		ok, _ := apiClient.EvaluateKey(fingerprint, device)
```
Comment (otavio) [inline ssh/server.go:250]: check `err` too.

File: ssh/ssh2ws.go [hunk 4]
```diff
@@ -55,6 +55,13 @@ func HandlerWebsocket(ws *websocket.Conn) {
 			return
 		}
 
+		if ok, err := apiClient.EvaluateKey(fingerprint, device); !ok {
+			fmt.Println(err)
+			ws.Write([]byte("Permission denied\r\n")) // nolint:errcheck
+			ws.Close()
+			return
+		}
+
```
Comment (otavio) [inline ssh/ssh2ws.go:64]: `ok` seems unused.

**Review by otavio — CHANGES_REQUESTED** [review]

File: api/routes/sshkeys.go (hunk 1 above)
Comment (eduardoveiga) [reply api/routes/sshkeys.go:146]: this way?

File: api/routes/sshkeys.go [hunk 5]
```diff
@@ -128,3 +129,24 @@ func CreatePrivateKey(c apicontext.Context) error {
...
+	if err != nil {
+		return err
+	}
+	return c.JSON(http.StatusOK, ok)
+
```
Comment (otavio) [inline api/routes/sshkeys.go:151]: The newline should be before the return, not after.

File: pkg/api/client/client_internal.go [hunk 6]
```diff
@@ -42,6 +43,19 @@ func (c *client) GetPublicKey(fingerprint, tenant string) (*models.PublicKey, er
 	return pubKey, nil
 }
 
+func (c *client) EvaluateKey(fingerprint string, dev *models.Device) (bool, error) {
+	var evaluate *bool
+	resp, _, errs := c.http.Post(buildURL(c, fmt.Sprintf("/internal/sshkeys/public-keys/evaluate/%s", fingerprint))).Send(dev).EndStruct(&evaluate)
+	if len(errs) > 0 {
+		return false, errs[0]
+	}
```
Comment (otavio) [inline pkg/api/client/client_internal.go:51]: Newline after this very first block.

**Review by otavio — CHANGES_REQUESTED** [review]

File: pkg/api/client/client_internal.go (hunk 6 above)
Comment (otavio) [reply pkg/api/client/client_internal.go:51]: I don't know if we're using the [`multierr`](https://github.com/uber-go/multierr) package. If not, we might look at it.

**Pull request merged**